File: citaria_backend/src/main/java/com/citaria/model/Servicio.java
package com.citaria.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;

import java.math.BigDecimal;
import java.util.Objects;

/**
 * Cada uno de los servicios que ofrece una organizacion
 */
@Entity
@Table(name = "servicio")
public class Servicio {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organizacion_id", nullable = false)
    private Organizacion organizacion;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "categoria_id")
    private Categoria categoria;

    @Column(nullable = false, length = 100)
    private String nombre;

    private String descripcion;

    @Column(name = "imagen_url", length = 500)
    private String imagenUrl;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal precio;

    @Column(name = "duracion_minutos", nullable = false)
    private Integer duracionMinutos;

    @Column(nullable = false)
    private Boolean activo = true;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Organizacion getOrganizacion() {
        return organizacion;
    }

    public void setOrganizacion(Organizacion organizacion) {
        this.organizacion = organizacion;
    }

    public Categoria getCategoria() {
        return categoria;
    }

    public void setCategoria(Categoria categoria) {
        this.categoria = categoria;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public String getImagenUrl() {
        return imagenUrl;
    }

    public void setImagenUrl(String imagenUrl) {
        this.imagenUrl = imagenUrl;
    }

    public BigDecimal getPrecio() {
        return precio;
    }

    public void setPrecio(BigDecimal precio) {
        this.precio = precio;
    }

    public Integer getDuracionMinutos() {
        return duracionMinutos;
    }

    public void setDuracionMinutos(Integer duracionMinutos) {
        this.duracionMinutos = duracionMinutos;
    }

    public Boolean getActivo() {
        return activo;
    }

    public void setActivo(Boolean activo) {
        this.activo = activo;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Servicio servicio)) return false;
        return Objects.equals(id, servicio.id) && Objects.equals(organizacion, servicio.organizacion) && Objects.equals(categoria, servicio.categoria) && Objects.equals(nombre, servicio.nombre) && Objects.equals(descripcion, servicio.descripcion) && Objects.equals(imagenUrl, servicio.imagenUrl) && Objects.equals(precio, servicio.precio) && Objects.equals(duracionMinutos, servicio.duracionMinutos) && Objects.equals(activo, servicio.activo);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, organizacion, categoria, nombre, descripcion, imagenUrl, precio, duracionMinutos, activo);
    }

    @Override
    public String toString() {
        return "Servicio{" +
                "id=" + id +
                ", organizacionId=" + (organizacion != null ? organizacion.getId() : null) +
                ", categoriaId=" + (categoria != null ? categoria.getId() : null) +
                ", nombre='" + nombre + '\'' +
                ", precio=" + precio +
                ", duracionMinutos=" + duracionMinutos +
                ", activo=" + activo +
                '}';
    }
}

File: citaria_frontend/lib/data/models/servicio.dart
class Servicio {
  final int? id;
  final int? organizacionId;
  final int? categoriaId;
  final String? nombreCategoria;
  final String nombre;
  final String? descripcion;
  final String? imagenUrl;
  final double precio;
  final int duracionMinutos;
  final bool? activo;

  const Servicio({
    this.id,
    this.organizacionId,
    this.categoriaId,
    this.nombreCategoria,
    required this.nombre,
    this.descripcion,
    this.imagenUrl,
    required this.precio,
    required this.duracionMinutos,
    this.activo,
  });

  factory Servicio.fromJson(Map<String, dynamic> json) {
    return Servicio(
      id: json['id'] == null ? null : (json['id'] as num).toInt(),
      organizacionId: json['organizacionId'] == null
          ? null
          : (json['organizacionId'] as num).toInt(),
      categoriaId: json['categoriaId'] == null
          ? null
          : (json['categoriaId'] as num).toInt(),
      nombreCategoria: json['nombreCategoria'] as String?,
      nombre: json['nombre'] as String,
      descripcion: json['descripcion'] as String?,
      imagenUrl: json['imagenUrl'] as String?,
      precio: (json['precio'] as num).toDouble(),
      duracionMinutos: (json['duracionMinutos'] as num).toInt(),
      activo: json['activo'] as bool?,
    );
  }

  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'categoriaId': categoriaId,
      'nombre': nombre,
      'descripcion': descripcion,
      'imagenUrl': imagenUrl,
      'precio': precio,
      'duracionMinutos': duracionMinutos,
      'activo': activo,
    };
  }
}
